yang/frr-zebra.yang

```diff
@@ -0,0 +1,1887 @@
+    }
+    leaf router-mac {
+      type yang:mac-address;
+      description
+        "The VNI Router MAC address.";
+    }
+    list vni-list {
+      description
+        "A list of the associated L2 VNIs.";
+      leaf vni-id {
+        type vni-id-type;
+        description
+          "An L2 VNI identifier.";
+      }
+    }
+  }
+
+  /*
+   * Debug options
+   */
+  grouping zebra-debugs {
+    leaf debug-events {
+      type empty;
+      description "Debug ZAPI events.";
+    }
+    leaf debug-zapi-send {
+      type empty;
+      description "Debug ZAPI messages sent.";
+    }
+    leaf debug-zapi-recv {
+      type empty;
+      description "Debug ZAPI messages received.";
+    }
+    leaf debug-zapi-detail {
+      type empty;
+      description "Debug ZAPI details.";
+    }
+    leaf debug-kernel {
+      type empty;
+      description "Debug kernel events.";
+    }
+    leaf debug-kernel-msg-send {
+      type empty;
+      description "Debug kernel messages sent.";
+    }
+    leaf debug-kernel-msg-recv {
+      type empty;
+      description "Debug kernel messages received.";
+    }
+    leaf debug-rib {
+      type empty;
+      description "Debug RIB processing.";
+    }
+    leaf debug-rib-detail {
+      type empty;
+      description "Debug RIB processing details.";
+    }
+    leaf debug-fpm {
+      type empty;
+      description "Debug the FIB Push Interface subsystem.";
+    }
+    leaf debug-nht {
+      type empty;
+      description "Debug Nexthop-tracking.";
+    }
+    leaf debug-nht-detail {
+      type empty;
+      description "Debug Nexthop-tracking details.";
+    }
+    leaf debug-mpls {
+      type empty;
+      description "Debug MPLS.";
+    }
+    leaf debug-vxlan {
+      type empty;
+      description "Debug VxLAN.";
+    }
+    leaf debug-pw {
+      type empty;
+      description "Debug pseudowires.";
+    }
+    leaf debug-dplane {
+      type empty;
+      description "Debug the dataplane subsystem.";
+    }
+    leaf debug-dplane-detail {
+      type empty;
+      description "Debug dataplane subsystem details.";
+    }
+    leaf debug-mlag {
+      type empty;
+      description "Debug MLAG.";
+    }
+  }
+
+  /*
+   * Main zebra container
+   */ 
+  container zebra {
+    description
+      "Data model for the Zebra daemon.";
+
+    leaf mcast-rpf-lookup {
+      type frr-zebra:mcast-rpf-lookup-mode;
+      default "mrib-then-urib";
+      description
+	"Multicast RPF lookup behavior.";
+    }
+
+    leaf ip-forwarding {
+      type boolean;
+      description
+	"IP forwarding status.";
+    }
+
+    leaf ipv6-forwarding {
+      type enumeration {
+	enum "unknown" {
+	  value -1;
+	  description
+	    "Unknown state.";
+	}
+	enum "off" {
+	  value 0;
+	  description
+	    "IPv6 forwarding disabled.";
+	}
+	enum "on" {
+	  value 1;
+	  description
+	    "IPv6 forwarding enabled.";
+	}
+      }
+      description
+	"IPv6 forwarding status.";
+    }
+
+    leaf workqueue-hold-timer {
```

Comment:
Since work-queues are used across several daemons, should we model a generic work-queue and allow it to be specified?